src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -134,11 +142,21 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
     this.setupSubscriptions()
     this.notify({type: 'initialize'})
 
+    this.initializeUtxoService()
```

Comment:
can this be done in constructor?

Reply:
Can we leverage DI here? `contructor(storage, utxoService)` so we can easily build fixtures? It's not the wallet implementation responsibility to instantiate the services but just consume them. It will avoid dealing with async issues, and it will allow UtxoService to not be null.

Reply:
but it might affect how the wallets are created right now,  it's too coupled :/, if it brings more complexity, let's do it in another iteration

Reply:
if you change getWalletImplementation to return the class, instead of an instance, then you can construct the instance with `id`

something like:
```
const Wallet = this.getWalletImplementation(walletImplementationId)
const wallet = new Wallet({storage, id: uuid.v4()})
```

Reply:
also, convert `_{methodName}` to `private {methodName}` as you go

Reply:
it needs a deep checking. I will do

Reply:
that's a good approach for this iteration, next: it's walletManager responsibility to set up the services/api/etc.. for a wallet, so it should build and pass (inject) everything that a wallet needs to work.